### PR TITLE
Added actual testdata

### LINKED ISSUE #51: Add better testdata
Current test files are 3 copies of the same file, limiting the expressiveness and possibilities of unit testing.

A real test dataset is required to assess functionality and cover more edge cases etc.

### PR DESCRIPTION
**Pull request by hechth**
Closes #51 

Added short sections of real LCMS QC files as testing data and expanded the tests for `extract_features`.
Added downloading of test files to the CI, similar to xmsannotator.

### CODE REVIEW

File: tests/testthat/test-extract_features.R [hunk 1]
```diff
@@ -1,33 +1,96 @@
...
+  patrick::cases(
+    RCX_shortened = list(
+      files = c("RCX_01_shortened_v2", "RCX_09_shortened_v2", "RCX_16_shortened_v2"),
+      tol = 1e-06,
+      min_pres = 0.7,
+      min_run = 4,
+      intensity_weighted = TRUE,
+      sd_cut = c(0.01, 500),
+      sigma_ratio_lim = c(0.01, 100)
+    )
```
Comment (xtrojak) [inline tests/testthat/test-extract_features.R:94]: Are we planning to add more test cases?

File: tests/testthat/test-prof.to.features.R [hunk 2]
```diff
@@ -1,11 +1,46 @@
...
+      sigma_ratio_lim = c(0.01, 100)
+    ),
+    RCX_16_shortened_v2 = list(
+      filename = c("RCX_16_shortened_v2_cdf.Rds"),
+      expected_filename = "RCX_16_shortened_v2_features.Rds",
+      sd_cut = c(0.01, 500),
+      sigma_ratio_lim = c(0.01, 100)
+    )
+  )
+)
```
Comment (xtrojak) [inline tests/testthat/test-prof.to.features.R]: newline

**Review by xtrojak — APPROVED** [review]
Looks good to me!

**Review by zargham-ahmad — APPROVED** [review]

File: tests/remote-files/RCX_01_shortened_v2.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+https://gitlab.ics.muni.cz/umsa/umsa-files/-/raw/aplcms_testdata/testdata/recetox-aplcms/RCX_01_shortened_v2/RCX_01_shortened_v2.mzML
+https://gitlab.ics.muni.cz/umsa/umsa-files/-/raw/aplcms_testdata/testdata/recetox-aplcms/RCX_01_shortened_v2/RCX_01_shortened_v2_cdf.Rds
+https://gitlab.ics.muni.cz/umsa/umsa-files/-/raw/aplcms_testdata/testdata/recetox-aplcms/RCX_01_shortened_v2/RCX_01_shortened_v2_features.Rds
```
Comment (xtrojak) [inline tests/remote-files/RCX_01_shortened_v2.txt]: Actually I noticed the links do not work properly. I think it should be 

```https://gitlab.ics.muni.cz/umsa/umsa-files/-/raw/master/testdata/recetox-aplcms/RCX_01_shortened_v2/RCX_01_shortened_v2.mzML```

(or maybe even permalink instead of `master`).

**Review by xtrojak — CHANGES_REQUESTED** [review]

File: tests/remote-files/RCX_01_shortened_v2.txt (hunk 3 above)
Comment (hechth) [reply tests/remote-files/RCX_01_shortened_v2.txt]: Ah yeah, the PR got merged - good point!

File: tests/testthat/test-extract_features.R (hunk 1 above)
Comment (hechth) [reply tests/testthat/test-extract_features.R:94]: Eventually - I mean it doesn't hurt to already have it in a parameterized form!

File: tests/testthat/test-prof.to.features.R (hunk 2 above)
Comment (hechth) [reply tests/testthat/test-prof.to.features.R]: solved in https://github.com/RECETOX/recetox-aplcms/pull/54/commits/feaf68fc6d67a4fda3130cb74f4f076d1bbdb632

File: tests/remote-files/RCX_01_shortened_v2.txt (hunk 3 above)
Comment (hechth) [reply tests/remote-files/RCX_01_shortened_v2.txt]: Solved in https://github.com/RECETOX/recetox-aplcms/pull/54/commits/8daaa7e0a7e79d4c8b91f356456c3724190f21c9
